spec/lib/pdf_filler/md502_pdf_spec.rb

```diff
@@ -0,0 +1,145 @@
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Enter social security number']).to eq("400000030")
+          expect(pdf_fields["Enter spouse&apos;s social security number"]).to eq("600000030")
+          expect(pdf_fields["Enter your first name"]).to eq("Mary")
+          expect(pdf_fields["Enter your middle initial"]).to eq("A")
+          expect(pdf_fields["Enter your last name"]).to eq("Lando")
+          expect(pdf_fields["Enter Spouse&apos;s First Name"]).to eq("Marty")
+          expect(pdf_fields["Enter Spouse&apos;s middle initial"]).to eq("B")
+          expect(pdf_fields["Enter Spouse&apos;s last name"]).to eq("Lando")
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Off"
+          expect(pdf_fields['Check Box - 3']).to eq "No"
+          expect(pdf_fields["MARRIED FILING Enter spouse&apos;s social security number"]).to eq("600000030")
+          expect(pdf_fields['Check Box - 4']).to eq "Off"
+          expect(pdf_fields['Check Box - 5']).to eq "Off"
+          expect(pdf_fields['6. Check here']).to eq "Off"
+        end
+      end
+    end
+
+    context "hoh" do
+      let(:intake) { create(:state_file_md_intake, :head_of_household) }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Off"
+          expect(pdf_fields['Check Box - 3']).to eq "Off"
+          expect(pdf_fields['Check Box - 4']).to eq "Yes"
+          expect(pdf_fields['Check Box - 5']).to eq "Off"
+          expect(pdf_fields['6. Check here']).to eq "Off"
+        end
+      end
+    end
+
+    context "qw" do
+      let(:intake) { create(:state_file_md_intake, :qualifying_widow) }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Off"
+          expect(pdf_fields['Check Box - 3']).to eq "Off"
+          expect(pdf_fields['Check Box - 4']).to eq "Off"
+          expect(pdf_fields['Check Box - 5']).to eq "Yes"
+          expect(pdf_fields['6. Check here']).to eq "Off"
+        end
+      end
+    end
+
+    context "dependent taxpayer" do
+      let(:intake) { create(:state_file_md_intake, :claimed_as_dependent) }
+
+      context "pulling fields from xml" do
+        it "sets other fields to the correct values" do
+          expect(pdf_fields['Check Box - 1']).to eq "Off"
+          expect(pdf_fields['Check Box - 2']).to eq "Off"
+          expect(pdf_fields['Check Box - 3']).to eq "Off"
+          expect(pdf_fields['Check Box - 4']).to eq "Off"
+          expect(pdf_fields['Check Box - 5']).to eq "Off"
+          expect(pdf_fields['6. Check here']).to eq "Yes"
```

Comment:
i have a question about this: it looks like we're currently relying on a field from the DF data (`primary_claim_as_dependent` in `direct_file_data`) to determine this but it also seems like that means their filing status itself is status "6". however, `state_file_base_intake` does not recognize filing statuses above 5. should we add to it? i think there are some product/program questions in here so happy for this to be a follow-up to this PR (if there's anything to be done)

Reply:
followed up in jira comments